lib/src/over_react_test/react_util.dart

```diff
@@ -481,8 +481,35 @@ Element queryByTestId(dynamic root, String value, {String key = defaultTestIdKey
 ///     </div>
 ///
 ///     queryAllByTestId(renderedInstance, 'value'); // returns both `inner` `<div>`s
-List<Element> queryAllByTestId(dynamic root, String value, {String key = defaultTestIdKey}) {
-  return findDomNode(root).querySelectorAll('[$key~="$value"]');
+List<Element> queryAllByTestId(dynamic root, String value, {String key = defaultTestIdKey, bool searchInShadowDom = false, int shadowDepth}) {
+  return _findAllDeep(findDomNode(root), _makeTestIdSelector(value, key: key), searchInShadowDom: searchInShadowDom, findMany: true, depth: shadowDepth);
+}
+
+String _makeTestIdSelector(String value, {String key = defaultTestIdKey}) => '[$key~="$value"]';
+
+dynamic /* Element | List<Element> */ _findAllDeep(Element root, String itemSelector, {bool searchInShadowDom = false, bool findMany = true, int depth}) {
+  List<Element> nodes = [];
+  var currentDepth = 0;
+  var lightElement = root.querySelector(itemSelector);
+  if (!findMany && lightElement != null){
+    return lightElement;
+  }
+  void recursiveSeek(dynamic _root) {
+    nodes = [...nodes, ..._root.querySelectorAll(itemSelector)];
+    if (!findMany && nodes.isNotEmpty) {
+      return;
+    }
+    // This is kinda gross but I figured this was only ever used in test situations so its probably alright.
+    if (depth == null || (depth != null && currentDepth < depth && searchInShadowDom == true)) {
```

Comment:
Maybe it would be better to short circuit before this if `searchInShadowDom` is false so this doesn't happen?
```suggestion
    if ((depth == null && searchInShadowDom == true) || (depth != null && currentDepth < depth && searchInShadowDom == true)) {
```

Reply:
```suggestion
    if ((depth == null || (depth != null && currentDepth < depth)) && searchInShadowDom == true) {
```

Reply:
👍